packages/js/src/ui/components/common/DropdownItem.tsx

```diff
@@ -0,0 +1,49 @@
+import { JSX, Show } from 'solid-js';
+import { useAppearance } from '../../context';
+import { AppearanceKey, cn, useStyle } from '../../helpers';
+import { Check } from '../../icons';
+
+export const DropdownItem = (props: {
+  label: string;
+  onClick: () => void;
+  isSelected?: boolean;
+  icon: () => JSX.Element;
+  appearanceKeyPrefix: string;
```

Comment:
⚠️ We shouldn't be passing appearance keys around in general. We need to make `DropdownItem` headless.

Reply:
Feel free to split this up into more components like it's done in radix: https://www.radix-ui.com/primitives/docs/components/dropdown-menu

I'd say `isSelected` can be done with a data attribute instead and just use css, and the children can be the label. Wdyt?
Final API could be
```
<Dropdown>
  <DropdownTrigger>
    //label and icon div
  </DropdownTrigger>
  <DropdownContent>
    <DropdownItem onClick class={/*add class and appearance keys as needed */}>
      //label and icon
    </DropdownItem>
  <DropdownContent/>
</Dropdown>
```
If we need to reuse the same styles multiple times, I'd say just export a class string from this file, and use it when necessary.
I.e.
```
<DropdownItem class={style(['dropdownItem', 'dropdownItem__moreActions'], cn(dropdownItemDefaultStyles, your_extra_classes))} />
```